From 7a84602297d36617dbdadeba55a2567031e5165b Mon Sep 17 00:00:00 2001
From: Jeff Layton <jlayton@kernel.org>
Date: Tue, 19 Mar 2024 12:34:45 -0400
Subject: [PATCH] 9p: explicitly deny setlease attempts
Git-commit: 7a84602297d36617dbdadeba55a2567031e5165b
Patch-mainline: v6.9-rc5
References: git-fixes

9p is a remote network protocol, and it doesn't support asynchronous
notifications from the server. Ensure that we don't hand out any leases
since we can't guarantee they'll be broken when a file's contents
change.

Signed-off-by: Jeff Layton <jlayton@kernel.org>
Signed-off-by: Eric Van Hensbergen <ericvh@kernel.org>
Acked-by: Anthony Iliopoulos <ailiop@suse.com>

---
 fs/9p/vfs_file.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/9p/vfs_file.c
+++ b/fs/9p/vfs_file.c
@@ -669,6 +669,7 @@ const struct file_operations v9fs_cached
 	.splice_read = generic_file_splice_read,
 	.splice_write = iter_file_splice_write,
 	.fsync = v9fs_file_fsync,
+	.setlease = simple_nosetlease,
 };
 
 const struct file_operations v9fs_cached_file_operations_dotl = {
@@ -737,4 +738,5 @@ const struct file_operations v9fs_mmap_f
 	.splice_read = generic_file_splice_read,
 	.splice_write = iter_file_splice_write,
 	.fsync = v9fs_file_fsync_dotl,
+	.setlease = simple_nosetlease,
 };
